diff --git a/m4/efl.m4 b/m4/efl.m4
index d051155e56..4b91c60963 100644
--- a/m4/efl.m4
+++ b/m4/efl.m4
@@ -22,14 +22,14 @@ dnl     dev_version = development version (svn revision).
 dnl     def_build_profile = dev or release based on 'dev' release parameter.
 AC_DEFUN([EFL_VERSION],
 [dnl
-m4_define([v_maj], [$1])dnl
-m4_define([v_min], [$2])dnl
-m4_define([v_mic], [$3])dnl
-m4_define([dev_version], m4_esyscmd([(git rev-list --count HEAD 2>/dev/null || echo 0) | tr -d '\n']))dnl
+m4_define([v_maj], [1])dnl
+m4_define([v_min], [99])dnl
+m4_define([v_mic], [99])dnl
+m4_define([dev_version], [0])
 m4_define([v_rev], m4_if($4, dev, [dev_version], [0]))dnl
 m4_define([v_rel], [])dnl
 m4_define([def_build_profile], m4_if($4, dev, [dev], [release]))dnl
-m4_define([efl_version], m4_if($4, dev, [v_maj.v_min.v_mic.v_rev], [v_maj.v_min.v_mic]))dnl
+m4_define([efl_version], [v_maj.v_min.v_mic])dnl
 dnl m4_define([efl_version], [v_maj.v_min.v_mic])dnl
 ])
 
